source/connection.c

```diff
@@ -431,6 +432,15 @@ void aws_http_connection_release(struct aws_http_connection *connection) {
     }
 }
 
+struct aws_server_user_data {
+    struct aws_allocator *alloc;
+    struct aws_http_server *server;
+    struct aws_mutex mutex;
+    struct aws_condition_variable condition_variable;
+    int setup_error_code;
+    bool setup_complete;
```

Comment:
This is a perfect use case for [aws_future](https://github.com/awslabs/aws-c-io/blob/318f7e57e7871e5b0d48a281cc5dcb7f79ccecdd/include/aws/io/future.h), which uses a mutex, condition variable, error code, and optional "success" value, plus a bunch of helper functions that can do exactly this kind of thing
```suggestion
    struct aws_future_void *setup_future;
```